addons/website_payment/static/src/snippets/s_donation/donation_snippet.js

```diff
@@ -0,0 +1,163 @@
+        this.defaultAmount = this.el.dataset.defaultAmount;
+        if (!!this.rangeSliderEl) {
+            this.rangeSliderEl.value = this.defaultAmount;
+            this.setBubble();
+        }
+    }
+
+    async willStart() {
+        cachedCurrency = cachedCurrency || this.waitFor(rpc("/website/get_current_currency"));
+        this.currency = await cachedCurrency;
+    }
+
+    start() {
+        const currencyEls = this.el.querySelectorAll('.s_donation_currency');
+        for (const currencyEl of currencyEls) {
+            currencyEl.remove();
+        }
+
+        const prefilledButtonEls = this.el.querySelectorAll('.s_donation_btn, .s_range_bubble');
+        for (const prefilledButtonEl of prefilledButtonEls) {
+            const insertBefore = this.currency.position === "before";
+            const currencySymbol = document.createElement('span');
+            currencySymbol.innerText = this.currency.symbol;
+            currencySymbol.classList.add('s_donation_currency', insertBefore ? "pe-1" : "ps-1");
+            this.insert(currencySymbol, prefilledButtonEl, insertBefore ? "afterbegin" : "beforeend");
+        }
+
+        const customButtonEl = this.el.querySelector("#s_donation_amount_input");
+        if (customButtonEl) {
+            this.registerCleanup(() => { customButtonEl.style.maxWidth = "" });
+            const canvasEl = document.createElement("canvas");
+            const context = canvasEl.getContext("2d");
+            context.font = window.getComputedStyle(customButtonEl).font;
+            const width = context.measureText(customButtonEl.placeholder).width;
+            customButtonEl.style.maxWidth = `${Math.ceil(width) + CUSTOM_BUTTON_EXTRA_WIDTH}px`;
+        }
+    }
+
+    setBubble() {
+        const bubbleEl = this.el.querySelector('.s_range_bubble');
+        const val = rangeSliderEl.value;
+        const min = rangeSliderEl.min || 0;
+        const max = rangeSliderEl.max || 100;
+        const newVal = Number(((val - min) * 100) / (max - min));
+        const tipOffsetLow = 8 - (newVal * 0.16); // the range thumb size is 16px*16px. The '8' and the '0.16' are related to that 16px (50% and 1% of 16px)
+
+        for (const child of bubbleEl.childNodes) {
+            if (child.nodeType === 3) {
+                child.nodeValue = val;
+            }
+        }
+        // Sorta magic numbers based on size of the native UI thumb (source: https://css-tricks.com/value-bubbles-for-range-inputs/)
+        bubbleEl.style.left = `calc(${newVal}% + (${tipOffsetLow}px))`;
+    }
+
+    /**
+     * @param {Event} ev
+     */
+    onClickPrefilled(ev) {
+        this.activeButtonEl = ev.currentTarget;
+        if (this.rangeSliderEl) {
+            this.rangeSliderEl.value = this.activeButtonEl.dataset.donationValue;
+            this.setBubble();
+        }
+    }
+
+    /**
+     * @param {Event} ev
+     */
+    onClickDonate(ev) {
+        const alertDangerEls = this.el.querySelectorAll('.alert-danger');
```

Comment:
I don't think you need a `querySelectorAll` here. The alert is only added inside this method, so there shouldn't be more than one at a time when you reach this line.